File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {

        unordered_set<char> vowels = {'a', 'e', 'i', 'o', 'u'};

        int l = 0;
        int r = 0;
        int count = 0;
        int large = 0;

        while(r<s.size())
        {
            if(vowels.count(s[r])) count ++;

            if(r-l+1==k)
            {   
                large = max(large,count);
                if(vowels.count(s[l])) count--;

                l++;
            }

            r++;
        }

        return large;
        
    }
};

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:

        count = 0 
        max_count = 0
        vowels = "aeiou"


        for i in range(0,k):
            if s[i] in vowels:
                count+=1

        max_count = count

        for i in range(k, len(s)):
            if (s[i]) in vowels:
                count+=1

            if (s[i-k]) in vowels:
                count-=1

            max_count = max(max_count,count)

        
        return max_count


        
        